addons/roe/roe.lua

```diff
@@ -0,0 +1,280 @@
+            cancel_roe(id)
+            coroutine.sleep(.5)
+        end
+    end
+end
+
+local true_strings = S{'true','t','y','yes','on'}
+local false_strings = S{'false','f','n','off'}
+local bool_strings = true_strings:union(false_strings)
+
+local function handle_setting(setting,val)
+    setting = setting and setting:lower() or setting
+    val = val and val:lower() or val
+    
+    if not setting or not settings:containskey(setting) then
+        error('specified setting (%s) does not exist':format(setting or ''))
+    elseif type(settings[setting]) == "boolean" then
+        if not val or not bool_strings:contains(val) then
+            settings[setting] = not settings[setting]
+        elseif true_strings:contains(val) then
+            settings[setting] = true
+        else    
+            settings[setting] = false
+        end
+            
+        notice('%s setting is now %s':format(setting, tostring(settings[setting])))
+    end
+
+end
+
+local function blacklist(add_remove,id)
+    add_remove = add_remove and add_remove:lower()
+    id = id and tonumber(id)
+
+    
+    if add_remove and id then
+        if add_remove == 'add' then
+            settings.blacklist:add(id)
+            notice('roe quest %d added to the blacklist':format(id))
+        elseif add_remove == 'remove' then
+            settings.blacklist:remove(id)
+            notice('roe quest %d removed from the blacklist':format(id))
+        else
+            error('`blacklist` specify \'add\' or \'remove\'')
+        end
+    else
+        error('`blacklist` requires two args, [add|remove] <quest id>')
+    end
+    
+end
+
+
+local function help()
+    notice([[ROE - Command List:
+1. help - Displays this help menu.
+2. save <profile name> : saves the currently set ROE to the named profile
+3. set <profile name> : attempts to set the ROE objectives in the profile
+    - Objectives may be canceled automatically based on settings.
+    - The default setting is to only cancel ROE that have 0 progress if space is needed
+4. unset : removes currently set objectives
+    - By default, this will only remove objectives without progress made
+5. settings <settings name> : toggles the specified setting
+    * settings:
+        * clear : removes objectives if space is needed (default true)
+        * clearprogress : remove objectives even if they have non-zero progress (default false)
+        * clearall : clears every objective before setting new ones (default false)
+6. blacklist [add|remove] <id> : blacklists a quest from ever being removed
+    - I do not currently have a mapping of quest IDs to names]])
+end
+
+local cmd_handlers = {
+    eval = eval,
+    save = save,
+    list = list,
+    set = set,
+    unset = unset,
+    setting = handle_setting,
+    blacklist = blacklist,
+    help = help,
+}
+
+
+local function inc_chunk_handler(id,data)
+    if id == 0x111 then
+        _roe.active:clear()
+        for i = 1,30 do

```

Comment:
The same `30` as above? I assume that's packet payload length, as indicated in `fields.lua`. Any reason not to use `packet.parse` here?

Reply:
The only reason was so I didn't have to put together a string with index i :(